### PR TITLE
Fix issues identified when parsing several hundred kicad 6 and 7 schematics

### PR DESCRIPTION
**Pull request by sAlexander**
In parsing several hundred schematics, I identified the following issues. I tried to keep my changes as clean and simple as possible, and I welcome any suggestions for improvement. For all of the changes, I included an example file that previously failed to parse and now successfully parses.

Also, thank you for sharing the kiutils library! It is a very well written and impactful library. I'm planning on using it more for my projects.

### CODE REVIEW

**Comment by sAlexander** [comment]
Full disclosure: I'm least happy with the [Fix issue with escaped backslash before string termination.](https://github.com/mvnmgrx/kiutils/pull/73/commits/b785b7f1ef183604fe75969fb02ed7bb2f789c25) change. While I fixed the case where "\\" parsed incorrectly, I suspect some other things will still parse incorrectly (e.g. "\\\"). I encourage you to give that change extra scrutiny! :)

File: src/kiutils/utils/sexpr.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
         (?P<brackl>\()|
         (?P<brackr>\))|
         (?P<num>[+-]?\d+\.\d+(?=[\ \)])|\-?\d+(?=[\ \)]))|
-        (?P<sq>"([^"]|(?<=\\)")*")|
+        (?P<sq>"([^"]|(?<=(?:\\\\|.[^\\])\\)")*")|
```
Comment (mvnmgrx) [inline src/kiutils/utils/sexpr.py:14]: As you said, this does only match your one problem case. I took a look into this and it seems like by just making the innter unnamed capture goup a none-capturing group, it parses every combination of \\\ and "\ just fine:
```
(?P<sq>"(?:[^"]|(?<=\\)")*")|
```

Something like this now works fine:
```   
(property "Fo|\\\\\\"\\"\"\\\\\\ot\"print" "some:part" (id 2) (at 106.68 683.895 0)
  (effects (font (size 1.27 1.27)) hide)
)
```

The official KiCad Library Utils also changes something here as well since i reused their S-Expression parser. It now does it the same way: https://gitlab.com/kicad/libraries/kicad-library-utils/-/blob/master/common/sexpr.py#L16

Please replace this line with the regex above and try parsing your problem file again if it still works as expected. Our unittests still work fine with this change on my local machine.

**Review by mvnmgrx — CHANGES_REQUESTED** [review]

**Comment by mvnmgrx** [comment]
Hello @sAlexander 

Thanks for taking the time to test `kiutils` so thoroughly. I've marked some minor changes in my review. Happy to merge your PR afterwards!

**Comment by mvnmgrx** [comment]
After some digging, i found out that the updated regex does not work correctly in every case (it even does not work correctly for the error case `"|\\"`). Here's what i came up with:

I've tried expanding the regex for quoted strings to the following:
```
(?P<sq>"(?:[^"]|(?<=\\)")*"(?:(?=\))|(?=\s)))
```

The `term_regex` would then look like this:
```python
term_regex = r'''(?mx)
    \s*(?:
        (?P<brackl>\()|
        (?P<brackr>\))|
        (?P<num>[+-]?\d+\.\d+(?=[\ \)])|\-?\d+(?=[\ \)]))|
        (?P<sq>"(?:[^"]|(?<=\\)")*"(?:(?=\))|(?=\s)))|
        (?P<s>[^(^)\s]+)
       )'''
```

It now checks for the closing quote correctly, regardless if the line ends or a closing tag is present. This works without disturbing the other parts of the S-Expression regex as it uses lookahead statements which will not be captured. 

I've updated my review comments with this new version of the regex.

I've also added a unittest for this (which currently fails as expected on the master branch), which you may want to merge into your branch for testing, here:
https://github.com/mvnmgrx/kiutils/blob/master/tests/test_misc.py#L25-L35
https://github.com/mvnmgrx/kiutils/blob/master/tests/testdata/misc/test_quotesAndBackslashInSexpr

Happy to hear your thoughts about this.

**Comment by mvnmgrx** [comment]
I've also added a simple unittest for the symbol ID edge cases:
https://github.com/mvnmgrx/kiutils/blob/master/tests/test_symbol.py#L124-L134
https://github.com/mvnmgrx/kiutils/blob/master/tests/testdata/symbol/test_symbolIdParser

**Comment by mvnmgrx** [comment]
I'm merging your PR and doing further work on your changes. Thanks for your contribution!

**Comment by sAlexander** [comment]
Sorry for going radio silent -- I don't check my github messages nearly often enough! And thanks for merging in the changes (with all of your fixes and tests).

I'm looking forward to using this project more!
